source/lensIndex.js

```diff
@@ -7,6 +7,8 @@ import update from './update.js';
 /**
  * Returns a lens whose focus is the specified index.
  *
+ * When `idx < 0 || idx >= list.length`, `R.set` or `R.over`, the original list is returned.
```

Comment:
Same comment about `idx` as above. Also `R.over` may throw depending on what the function is